### ISSUE TITLE
Encountering "mount.nfs: access denied by server" on Ubuntu 16.10 with NFS synced_folder configured

### ISSUE THREAD

**Issue by revprez** [mitchellh/vagrant#8241]
## System

1. Ubuntu 16.10
2. nfs-kernel-server, nfs-common, and rpcbind are installed
3. Virtualbox 5.1.6
4. Vagrant 1.9.1
5. Also employing [vagrant-bindfs](https://github.com/gael-ian/vagrant-bindfs) (though error proceeds its invocation)
6. Using [centos/7](https://atlas.hashicorp.com/centos/boxes/7) box.
7. `/etc/exports` modified by Vagrant like so:  
	```
	# VAGRANT-BEGIN: 1000 4b162241-d3a5-410c-a910-09a83ea90146
	"/home/revprez/dev/scm/git/encirca/orchestration/encirca-core-orchestration" 192.168.50.4(rw,no_subtree_check,all_squash,anonuid=1000,anongid=1000,fsid=4169747921)
	# VAGRANT-END: 1000 4b162241-d3a5-410c-a910-09a83ea90146
	```
	

## Issue

After running `vagrant up`,  encountering the following error:
```
Jan 31 11:06:55 mirai systemd[1]: Starting NFS server and services...
Jan 31 11:06:55 mirai systemd[1]: Started NFS server and services.
==> default: Mounting NFS shared folders...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3,udp 192.168.50.1:/home/revprez/dev/scm/git/encirca/orchestration/encirca-core-orchestration /vagrant-nfs
result=$?
if test $result -eq 0; then
if test -x /sbin/initctl && command -v /sbin/init && /sbin/init 2>/dev/null --version | grep upstart; then
/sbin/initctl emit --no-wait vagrant-mounted MOUNTPOINT=/vagrant-nfs
fi
else
exit $result
fi


Stdout from the command:



Stderr from the command:

mount.nfs: access denied by server while mounting 192.168.50.1:/home/revprez/dev/scm/git/encirca/orchestration/encirca-core-orchestration
```

## Investigation

1. Logged into the guest (`vagrant ssh`)
1. Ran the NFS mount command with some verbosity and observed the output.
	```
	[vagrant@regystar ~]$ sudo mount -v -o vers=3,udp 192.168.50.1:/home/revprez/dev/scm/git/encirca/orchestration/encirca-core-orchestration /vagrant-nfs
	mount.nfs: timeout set for Tue Jan 31 17:36:07 2017
	mount.nfs: trying text-based options 'vers=3,udp,addr=192.168.50.1'
	mount.nfs: prog 100003, trying vers=3, prot=17
	mount.nfs: trying 192.168.50.1 prog 100003 vers 3 prot UDP port 2049
	mount.nfs: prog 100005, trying vers=3, prot=17
	mount.nfs: trying 192.168.50.1 prog 100005 vers 3 prot UDP port 45228
	mount.nfs: mount(2): Permission denied
	mount.nfs: access denied by server while mounting 192.168.50.1:/home/revprez/dev/scm/git/encirca/orchestration/encirca-core-orchestration

	```
## Related issues

Mine my duplicate #5424, #100, and others.

**Comment by kikitux** [comment]
@revprez 

Can you please:

- provide `/var/log/message` or relevant file for ubuntu (syslog?)
- try to mount a folder outside /home, try /media/test 

sudo mkdir /media/test
sudo chown <user> /media/test

and share that folder

Thanks

**Comment by revprez** [comment]
Same issue:

```
Jan 31 16:56:49 mirai systemd[1]: Starting NFS server and services...
Jan 31 16:56:50 mirai systemd[1]: Started NFS server and services.
==> default: Mounting NFS shared folders...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3,udp 192.168.50.1:/media/test /vagrant-nfs
result=$?
if test $result -eq 0; then
if test -x /sbin/initctl && command -v /sbin/init && /sbin/init 2>/dev/null --version | grep upstart; then
/sbin/initctl emit --no-wait vagrant-mounted MOUNTPOINT=/vagrant-nfs
fi
else
exit $result
fi


Stdout from the command:



Stderr from the command:

mount.nfs: access denied by server while mounting 192.168.50.1:/media/test
```

**Comment by revprez** [comment]
Additionally, tried manually with some verbosity on the guest:
```
[vagrant@regystar ~]$ sudo mount -v -o vers=3,udp 192.168.50.1:/media/test/ /vagrant-nfs
mount.nfs: timeout set for Wed Feb  1 04:53:14 2017
mount.nfs: trying text-based options 'vers=3,udp,addr=192.168.50.1'
mount.nfs: prog 100003, trying vers=3, prot=17
mount.nfs: trying 192.168.50.1 prog 100003 vers 3 prot UDP port 2049
mount.nfs: prog 100005, trying vers=3, prot=17
mount.nfs: trying 192.168.50.1 prog 100005 vers 3 prot UDP port 37853
mount.nfs: mount(2): Permission denied
mount.nfs: access denied by server while mounting 192.168.50.1:/media/test/
```

**Comment by revprez** [comment]
Also confirmed this fails on macOS Sierra.  Even in what I think should be the trivial case:

```
Vagrant.configure("2") do | config |

   config.vm.box = 'centos/7'
   config.vm.hostname = hostname

    config.vm.network :private_network, ip: "192.168.50.4"

    config.vm.synced_folder ".", "/vagrant-nfs", type: "nfs"
    # config.bindfs.bind_folder "/vagrant-nfs", "/vagrant"
	
    

end
```

**Comment by peterdragon** [comment]
See also https://github.com/mitchellh/vagrant/issues/8138

**Comment by Abam** [comment]
There are some problems when NAT is enabled in virtualbox.  Instead of setuping network for eth1, Vagrant try to configure eth0 which is alreadry used.

Please check /etc/network/interfaces file for confirmation.

**Comment by revprez** [comment]
@kikitux : Not sure if you saw my previous replies, but after looking at #8138 at @peterdragon's suggestion, I note that mounting succeeds after restarting the network service:

```
[root@regystar ~]# ip route
default via 10.0.2.2 dev eth0  proto static  metric 100
10.0.2.0/24 dev eth0  proto kernel  scope link  src 10.0.2.15  metric 100
[root@regystar ~]# service network restart
Restarting network (via systemctl):                        [  OK  ]
[root@regystar ~]# ip route
default via 10.0.2.2 dev eth0  proto static  metric 100
10.0.2.0/24 dev eth0  proto kernel  scope link  src 10.0.2.15  metric 100
169.254.0.0/16 dev eth1  scope link  metric 1003
192.168.50.0/24 dev eth1  proto kernel  scope link  src 192.168.50.4
[root@regystar ~]# mount -o vers=3,udp 192.168.50.1:/Users/revprez/dev/scm/git/encirca/orchestration/encirca-core-orchestration /vagrant-nfs
```

However, this doesn't solve my problem, as I have additional bindfs configuration to proceed to afterwards. `vagrant reload` also blows away the new route.

**Comment by revprez** [comment]
@kikitux : This issue seems to be resolved in 96611341a96d7d19fdade5556a110b22c6add22b. A quick patch succeeded.

Any idea when the next Vagrant release is due?

```
From 96611341a96d7d19fdade5556a110b22c6add22b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C3=ABl=20Faille?= <michael@faille.io>
Date: Wed, 21 Dec 2016 20:31:53 -0500
Subject: [PATCH] Revert "Fix `service network restart` on RHEL-7 / Fedora"

`/etc/init.d/network restart` already restart NM and shutdown interfaces.

In start() :
```
        if [ "$(LANG=C nmcli -t --fields running general status 2>/dev/null)" = "running" ]; then
                nmcli connection reload
        fi

```

In stop() :

```
       for i in $vpninterfaces $xdslinterfaces $bridgeinterfaces $vlaninterfaces $remaining; do
                unset DEVICE TYPE
                (. ./ifcfg-$i
                if [ -z "$DEVICE" ] ; then DEVICE="$i"; fi

                if ! check_device_down $DEVICE; then
                   action $"Shutting down interface $i: " ./ifdown $i boot
                   [ $? -ne 0 ] && rc=1
                fi
                )
        done
```
Where $remaining include all "others" interfaces including eth*

This reverts commit 166d10d4e16836e908be0bd94c2f671cac2b38b4.
---
 plugins/guests/redhat/cap/change_host_name.rb   |  8 ++------
 plugins/guests/redhat/cap/configure_networks.rb | 12 ++++--------
 2 files changed, 6 insertions(+), 14 deletions(-)

diff --git a/plugins/guests/redhat/cap/change_host_name.rb b/plugins/guests/redhat/cap/change_host_name.rb
index e950c987b..afe461d4f 100644
--- a/plugins/guests/redhat/cap/change_host_name.rb
+++ b/plugins/guests/redhat/cap/change_host_name.rb
@@ -31,12 +31,8 @@ module VagrantPlugins
                 sed -i'' '1i 127.0.0.1\\t#{name}\\t#{basename}' /etc/hosts
               }
 
-              # Restart network (through NetworkManager if running)
-              if service NetworkManager status 2>&1 | grep -q running; then
-                service NetworkManager restart
-              else
-                service network restart
-              fi
+              # Restart network
+              service network restart
             EOH
           end
         end
diff --git a/plugins/guests/redhat/cap/configure_networks.rb b/plugins/guests/redhat/cap/configure_networks.rb
index 396457968..6749f1bdc 100644
--- a/plugins/guests/redhat/cap/configure_networks.rb
+++ b/plugins/guests/redhat/cap/configure_networks.rb
@@ -36,7 +36,7 @@ module VagrantPlugins
 
             # Add the new interface and bring it back up
             final_path = "#{network_scripts_dir}/ifcfg-#{network[:device]}"
-            commands << <<-EOH.gsub(/^ {14}/, '')
+            commands << <<-EOH.gsub(/^ */, '')
               # Down the interface before munging the config file. This might
               # fail if the interface is not actually set up yet so ignore
               # errors.
@@ -48,13 +48,9 @@ module VagrantPlugins
             EOH
           end
 
-          commands << <<-EOH.gsub(/^ {12}/, '')
-            # Restart network (through NetworkManager if running)
-            if service NetworkManager status 2>&1 | grep -q running; then
-              service NetworkManager restart
-            else
-              service network restart
-            fi
+          commands << <<-EOH.gsub(/^ */, '')
+            # Restart network
+            service network restart
           EOH
 
           comm.sudo(commands.join("\n"))
-- 
2.11.1
```

**Comment by chrisroberts** [comment]
Hi there,

It looks like this has been resolved within a previously shipped version of Vagrant so I am now closing this issue. If the original issue was not fully resolved, please reopen this issue or create a new one.

Cheers!
